core/src/main/java/io/grpc/ClientInterceptor.java

```diff
@@ -53,6 +53,10 @@
    * around the result of {@code next.newCall()}, whereas a simpler implementation may just modify
    * the header metadata prior to returning the result of {@code next.newCall()}.
    *
+   * <p>{@code next.newCall()} <strong>must not</strong> be called under a different {@link Context}
+   * other than the current {@code Context}. The outcome of such usage is undefined and may cause
```

Comment:
It's just defined by the implementation, but not what we really want. Calling it undefined means we don't support whatever the current implementation is doing and we are free to change it in the future.